bootstrap-salt.sh

```diff
@@ -4830,15 +4830,22 @@ install_amazon_linux_ami_2_git_deps() {
         yum -y install ca-certificates || return 1
     fi
 
-    PIP_EXE='pip'
-    if __check_command_exists python2.7; then
-        if ! __check_command_exists pip2.7; then
-            __yum_install_noinput python2-pip
+    if __check_command_exists python3; then
```

Comment:
When this check is being executed, it seems to be that Python3 is not yet installed on Amazon Linux.
I think you want to call `install_amazon_linux_ami_2_deps` before this (not afterwards) to make sure everything related to Py3 is in place.

Reply:
I am in two minds about that, since the user would need to have '-y -x python3' to install Python 3 versions and install **Python 3** itself as it currently is, but what if the user want to install the Python 3 support but not use it yet, that is, not install **Python 3** itself just yet.

Noting that **Python 3** is not automatically installed on Amazon Linux 2 ( that is Python 3.7.4) but that could change after this month, and also Amazon might follow after Redhat 7 (which it is based on bar Python 3.6) and have Python 3 support installed, as per Redhat 7.7.

Could cut both ways as to which is better for the user.  Any insight gladly accepted.

Reply:
In the end followed your suggestion